GoogleChromeLabs/ps-analysis-tool
# Two-step process for toggling CDP state

## Description
This PR aims to remove the automatic reload of tabs if any extension setting is changed. Instead, a message will be shown to the user to click on reload all the tabs for settings to take effect and to get accurate data. This prevents the user from reloading the tab and losing any data.

## Relevant Technical Choices
- Session storage is used to store the data temporarily.
- Until the page is reloaded from our mechanism, the PSAT shows the settings from session storage.
- Once the page is reloaded with our mechanism extension settings storage is set and the PSAT takes value from the extension storage.

## Testing Instructions
- Clone this branch and in the terminal run `npm start`.
- Go to any site and open the settings page. Switch any settings, and you should see a toast message at the bottom.
- You should see the same message in the popup.

## Screenshot/Screencast

https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/59614577/7138ae34-734e-4e8b-8507-4469272fc5dd




---

## Checklist
